changes/2020-07-15_encryption-context-reserved-prefix.md/change.md

```diff
@@ -0,0 +1,113 @@
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                               |
+| ------------------------------------------- |
+| [Structures](../../framework/structures.md) |
+| [Encrypt](../../client-apis/encrypt.md)     |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| Python     | [aws-encryption-sdk-python](https://github.com/aws/aws-encryption-sdk-python)         |
+| Java       | [aws-encryption-sdk-java](https://github.com/aws/aws-encryption-sdk-java)             |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The specification currently states that the encryption context
+MUST reserve the `aws-crypto-public-key` key field for use by the AWS Encryption SDK (ESDK)
+and SHOULD reserve any key field with the prefix `aws`.
+We will replace these two requirements with a statement that the encryption context
+MUST reserve any key field with the prefix `aws-crypto-`.
+
+We will also relocate this statement to apply to the encryption context passed to encrypt,
+rather than on all encryption context values.
+
+## Out of Scope
+
+It is possible that an existing encryption context used as input to encrypt
+will be broken by this change.
+Designing a migration approach for affected users
+is out of scope for this document.
+
+## Motivation
+
+The current statement using "SHOULD" is too weak to be of any use.
+Some users could be providing encryption contexts
+with key fields using the `aws` prefix,
+and in fact this is relatively likely
+since the context of many uses of the ESDK
+involves one or more relevant values related to AWS services.
+This means if any future versions of CMMs packaged with the ESDK
+choose to add additional internal key fields to the encryption context,
+they could be a breaking change.
+
+In addition, stating that the encryption context structure itself
+MUST reserve any set of key fields
+is questionable.
+Interpreted literally, this leads to a contradiction
+since the [Default CMM](../../framework/default-cmm.md) itself adds a key field with the reserved prefix.
```

Comment:
```suggestion
since the [Default CMM](https://github.com/awslabs/aws-encryption-sdk-specification/blob/2eb5b5fd81a1e8d57655efa3f73573b689e8cd42/framework/default-cmm.md) itself
adds a key field with the reserved prefix.
```

Reply:
See #137, ignoring :)